File: modules/layout/source/types/layout_types/func.ts
export class Func {
    id: string
    name: string // название функции
    type: ParamType // тип функции
    body: string // тело функции
    params: ParamType[] // параметры передаваемые функции
    used: Func[] // другие используемые функции
}

// Тип параметров
export enum ParamType {
    STRING = 'string', // строка
    VOID = 'void', // не возвращает
    NUMBER = 'number', // число
    ANY = 'any', // любой
}

File: modules/layout/app/types/layout_types/func.go
package layout_types
 
type Func  struct{
    Id string
    Name string // название функции
    Type ParamType // тип функции
    Body string // тело функции
    Params []ParamType // параметры передаваемые функции
    Used []Func // другие используемые функции
}

// Тип параметров
type ParamType string
const (
    STRING ParamType = "string" // строка
    VOID ParamType = "void" // не возвращает
    NUMBER ParamType = "number" // число
    ANY ParamType = "any" // любой
)
